### PR TITLE
Implement Ability to Support Custom Galaxy Server Settings

### PR DESCRIPTION
**Pull request by beeankha**
Addressing Issue https://github.com/ansible/ansible-builder/issues/79

### CODE REVIEW

File: ansible_builder/main.py [hunk 1]
```diff
@@ -307,6 +317,19 @@ def prepare_system_steps(self, bindep_output):
...
+    def prepare_final_stage_steps(self):
+        self.steps.append("FROM {0}".format(self.base_image))
+        return self.steps
+
+    def prepare_galaxy_copy_steps(self):
+        if self.definition.get_dep_abs_path('galaxy'):
+            self.steps.extend(GalaxyCopySteps())
+        return self.steps
+
+    def delete_intermediate_image(self):
```
Comment (shanemcd) [inline ansible_builder/main.py]: Not sure it makes sense to have this live in the `Containerfile` class. Maybe this should live somewhere in `AnsibleBuilder`?

Comment (beeankha) [reply ansible_builder/main.py]: Moving it to the bottom of AnsibleBuilder class for now.

File: ansible_builder/steps.py [hunk 2]
```diff
@@ -46,7 +46,21 @@ def __init__(self, requirements_naming):
...
+
+class GalaxyCopySteps(Steps):
+    def __init__(self):
+        """Assumes given requirements file name has been placed in the build context
+        """
+        self.steps = []
+        self.steps.extend([
+            "",
+            "COPY --from=builder {0} {0}".format(constants.base_roles_path),
+            "COPY --from=builder {0} {0}".format(constants.base_collections_path),
```
Comment (pabelanger) [inline ansible_builder/steps.py:63]: What is we copied in to the base image (from builder) the tarball folder of collections.   Then, we could run ansible-galaxy install tarballs/*.tar.gz again.

The main idea here is, there will be uscases where uses will build custom collections tarballs, and need to install them over something from the galaxy server. And people could remove the need for requirement files in the builder phase, and copy in their own tarballs.

This is kinda of how we are doing it with python wheels today in python-builder / python-base images: https://github.com/ansible/python-builder-image/ https://github.com/ansible/python-base-image/

Comment (shanemcd) [reply ansible_builder/steps.py:63]: I like this idea, but we may want to do this in a followup PR depending on whether @beeankha thinks she can accomplish this before she leaves for winter break. I think she was hoping to get something merged before then.

File: docs/definition.rst [hunk 3]
```diff
@@ -30,9 +34,18 @@ definition file in the ``base_image`` section as a string; this is an
...
 command line, the CLI option will take precedence.
 
+Ansible Config File Path
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+When using an ``ansible.cfg`` file to pass a token and other settings for a
+private account to an Automation Hub server, listing the config file path here
+(as a string) will enable it to be included as a build argument in the initial
+phase of the build. The information contained within the config file will not
+appear in the output of ``history`` when run against the final image.
```
Comment (AlanCoding) [inline docs/definition.rst]: but will be in the history of the intermediate image... so don't share that.

**Review by shanemcd — APPROVED** [review]

**Pull request merged**